File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/TestReport.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class TestReport : DomainResource, IResource
{
    public string? Tester { get; set; }
    public string? Name { get; set; }
    public required Base.ResourceReference TestScript { get; set; }
    public TestReport_Participant[]? Participant { get; set; }
    public TestReport_Setup? Setup { get; set; }
    public required string Status { get; set; }
    public required string Result { get; set; }
    public string? Score { get; set; }
    public Base.Identifier? Identifier { get; set; }
    public string? Issued { get; set; }
    public TestReport_Teardown? Teardown { get; set; }
    public TestReport_Test[]? Test { get; set; }

    public class TestReport_Participant : BackboneElement
    {
        public required string Uri { get; set; }
        public required string Type { get; set; }
        public string? Display { get; set; }
    }

    public class TestReport_Setup_Action_Assert : BackboneElement
    {
        public string? Detail { get; set; }
        public required string Result { get; set; }
        public string? Message { get; set; }
    }

    public class TestReport_Setup_Action_Operation : BackboneElement
    {
        public string? Detail { get; set; }
        public required string Result { get; set; }
        public string? Message { get; set; }
    }

    public class TestReport_Setup_Action : BackboneElement
    {
        public TestReport_Setup_Action_Assert? Assert_ { get; set; }
        public TestReport_Setup_Action_Operation? Operation { get; set; }
    }

    public class TestReport_Setup : BackboneElement
    {
        public required TestReport_Setup_Action[] Action { get; set; }
    }

    public class TestReport_Teardown_Action : BackboneElement
    {
        public required Base.ResourceReference Operation { get; set; }
    }

    public class TestReport_Teardown : BackboneElement
    {
        public required TestReport_Teardown_Action[] Action { get; set; }
    }

    public class TestReport_Test_Action : BackboneElement
    {
        public Base.ResourceReference? Assert_ { get; set; }
        public Base.ResourceReference? Operation { get; set; }
    }

    public class TestReport_Test : BackboneElement
    {
        public string? Name { get; set; }
        public required TestReport_Test_Action[] Action { get; set; }
        public string? Description { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/TestReport.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class TestReportParticipant(BackboneElement):
    uri: str
    type: str
    display: Optional[str] = None

@dataclass(kw_only=True)
class TestReportSetupActionAssert(BackboneElement):
    detail: Optional[str] = None
    result: str
    message: Optional[str] = None

@dataclass(kw_only=True)
class TestReportSetupActionOperation(BackboneElement):
    detail: Optional[str] = None
    result: str
    message: Optional[str] = None

@dataclass(kw_only=True)
class TestReportSetupAction(BackboneElement):
    assert_: Optional[TestReportSetupActionAssert] = None
    operation: Optional[TestReportSetupActionOperation] = None

@dataclass(kw_only=True)
class TestReportSetup(BackboneElement):
    action: list[TestReportSetupAction] = field(default_factory=list)

@dataclass(kw_only=True)
class TestReportTeardownAction(BackboneElement):
    operation: Reference

@dataclass(kw_only=True)
class TestReportTeardown(BackboneElement):
    action: list[TestReportTeardownAction] = field(default_factory=list)

@dataclass(kw_only=True)
class TestReportTestAction(BackboneElement):
    assert_: Optional[Reference] = None
    operation: Optional[Reference] = None

@dataclass(kw_only=True)
class TestReportTest(BackboneElement):
    name: Optional[str] = None
    action: list[TestReportTestAction] = field(default_factory=list)
    description: Optional[str] = None

@dataclass(kw_only=True)
class TestReport(DomainResource):
    tester: Optional[str] = None
    name: Optional[str] = None
    test_script: Reference
    participant: Optional[List[TestReportParticipant]] = field(default_factory=list)
    setup: Optional[TestReportSetup] = None
    status: str
    result: str
    score: Optional[float] = None
    identifier: Optional[Identifier] = None
    issued: Optional[str] = None
    teardown: Optional[TestReportTeardown] = None
    test: Optional[List[TestReportTest]] = field(default_factory=list)